### PR TITLE
Full features

### PR DESCRIPTION
**Pull request by Oluwadamilareolusakin**
# Description
This PR implements all the features of the battleship game. It comprises of a game, coordinates generator, ship, board and prompts modules.

### CODE REVIEW

**Comment by Oluwadamilareolusakin** [comment]
@bellom Thank you for the review :+1:. I'm currently working alone as my last partner left the program for personal reasons :slightly_smiling_face:. I will look over the changes requested, and make appropriate changes, thank you! :rocket:

**Comment by Oluwadamilareolusakin** [comment]
All changes have been made and tests were written for all factories except those that manipulate the DOM as they are irrelevant to test.

File: package.json [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  "devDependencies": {
+    "clean-webpack-plugin": "^3.0.0",
+    "css-loader": "^3.2.1",
+    "html-webpack-plugin": "^3.2.0",
+    "node-sass": "^4.13.0",
+    "sass-loader": "^8.0.0",
+    "style-loader": "^1.0.1",
+    "webpack": "^4.41.2",
+    "webpack-cli": "^3.3.10"
+  }
```
Comment (jofer86) [inline package.json]: **[REQUIRED]** No linter configuration found on your package.json file, please do refer to my top comment to get the full scope.

File: .gitignore [hunk 2]
```diff
@@ -0,0 +1 @@
+node_modules
```
Comment (jofer86) [inline .gitignore]: Good work addressing this: 
image

**Comment by Oluwadamilareolusakin** [comment]
Please see this screenshot below as proof of no stickler errors. For some reason it's glitching on the repo so I had to go do the review directly on the stickler-ci site
screenshot

**Pull request merged**